sklearn/tree/_criterion.pyx

```diff
@@ -230,10 +261,19 @@ cdef class ClassificationCriterion(Criterion):
         self.start = 0
         self.pos = 0
         self.end = 0
+        self.missing_direction = 2  # Ignore the missing values by default
 
         self.n_outputs = n_outputs
+
+        self.n_missing = 0
+        self.weighted_n_missing = 0
+
+        self.n_nonmissing = 0
+        self.weighted_n_nonmissing = 0
```

Comment:
I would prefer calculating it on the fly, since it's just a single subtraction. Shouldn't the examples which don't exist not influence the impurity calculation?